### ISSUE TITLE
Public workstream Pilot

### ISSUE THREAD

**Issue by hurtstotouchfire** [edx/open-source-process-wg#83]
The purpose of this is to attempt to externalize some of our prioritization and workload and to create an intake funnel for contributions from outside the team. Tooling wise, here’s what I think this involves:

- [ ] Public workstream tracking boards in github for each participating team / project to aggregate:

  - Funded contributions
  - Blended development
  - Operations work (deprs, devops, dependency updates, and open edx release tracking)
  - Known bugs
  - High level (epics) feature roadmap for internal 2U work

- [ ] A jira <-> GH integration to streamline issue tracking and sync anything that’s duplicated across Jira and the public board (@nedbat is bottom-lining this piece).

MVP functions:

  - Create Jira ticket matching GH issue, on demand
  - Create matching GH issue from Jira issue, on demand
  - Designated links between issues in both systems
  - Display GH issue status on matching issue within Jira

Use Cases:

  - Jira tickets as action items from GH issues
    - Purpose: notice for internal workstream interruptions
    - Support many Jiras to one GH issue
    - Only sync status
    - GH Issue is source of truth / main conversation point
  - Parallel tracking tickets
    - Purpose: visibility for community
    - 1:1 ticket relationship
    - Most likely for Epics only
    - Useful to have attributes synced over time
    - Jira is source of truth / main conversation point

- [ ] Public wiki space for approach docs, roadmap related docs, any further context needed to understand the work in flight or planned. (see [#476](https://github.com/openedx/tcril-engineering/issues/476) for thought-work around this)

**Comment by hurtstotouchfire** [comment]
We have a third use case we're trying to flesh out, @nedbat can you comment about it?

**Comment by jristau1984** [comment]
Is "everyone else (also GitHub-centric)" known, or an assumption?  I know that RaccoonGang uses YouTrack internally, and OpenCraft uses Jira internally.  I presume "GitHub centric" is our avenue for seeing into their work, not their preferred way of tracking.

**Comment by antoviaque** [comment]
It's great to see this initiative, thank you for leading it! We all do need to expose more of our workflows and task management to be able to collaborate more effectively, and I'd love to find an approach for OpenCraft to participate in the effort.

We mainly use Jira (internally) and Gitlab (externally) for our task tracking, but given that Open edX is currently centered around Github, it makes sense to me to try to have the Open edX public workstreams there.

@bradenmacdonald Do you think we could try this out with blockstore, and the upcoming projects around it with modular learning?

@itsjeyd Any comments on your side? Since we already expose our work through OSPRs tickets & updates, would it be worth considering this as part of the work Michelle and you are planning, to organize upstream reviews?

**Comment by nedbat** [comment]
Grooming: split checkboxes into sub issues.

**Comment by bradenmacdonald** [comment]
@antoviaque Sure, once the Jira->GitHub syncing mechanism is done, I'm sure we could use it to help make our tasks more public on those projects. I'm not sure it would add too much value though, as it's more like "what we're currently working on" than a roadmap of what we're going to do over the longer term.

**Comment by itsjeyd** [comment]
@antoviaque Yep, the process changes proposed here definitely relate to the work that @mphilbrick211 and I started doing as part of the [OSPR Project Manager|https://openedx.atlassian.net/wiki/spaces/COMM/pages/3548807177/Community+Contributions+Project+Manager] role.

The prospect of having more visibility into how participating teams are prioritizing tasks seems especially relevant, as it would probably help us identify PRs for which to accelerate the review process. (I had already raised [a question](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3548807177/Community+Contributions+Project+Manager?focusedCommentId=3555131420) about that particular point on the wiki.)

It would also be useful to be able to point PR authors to public boards that they could check/follow themselves in order to get updates on when to expect a review for their PR. (Based on what I've seen so far, PR authors currently depend on OSPR Project Managers for that kind of information; and [it's not always possible](https://github.com/openedx/frontend-app-learning/pull/977#issuecomment-1267355840) to answer their questions.)

**Comment by antoviaque** [comment]
@itsjeyd Thank you! :+1: 

@bradenmacdonald 

> Sure, once the Jira->GitHub syncing mechanism is done, I'm sure we could use it to help make our tasks more public on those projects

I'm not sure I would start with an automation step, at least on our side -- besides delaying the experiment (waiting for the tool/dev), it usually means that we automate something we don't really know how/if it works. I'm a big fan of starting processes with the human element, even if that means doing  few things manually at first (and once it works, _then_ automate it).

Couldn't we reuse the approach we are currently using for our own free software projects, like Listaflow, but with Github instead of Gitlab? Ie, create the tickets related to those projects on Github/gitlab _first_, and then only link to it from our internal tickets? The discussions can then also fully happen publicly.

> I'm not sure it would add too much value though, as it's more like "what we're currently working on" than a roadmap of what we're going to do over the longer term.

Since the "what we are currently working on" affects others and generates upstream changes, it's important to have this information public imho -- otherwise it's like trying to contribute to a project where the task tracker is private?

**Comment by nedbat** [comment]
I've made a separate issue for designing the behavior of the integration: #86

**Comment by bradenmacdonald** [comment]
@antoviaque 

>  Ie, create the tickets related to those projects on Github/gitlab first, and then only link to it from our internal tickets? The discussions can then also fully happen publicly.

Sorry, I think I am fundamentally confused about what is even being asked here. We are tracking the Modular Learning project on GitHub already, on this board: https://github.com/orgs/openedx/projects/4/views/14 So what is expected from this initiative that's different than what we're already doing?

**Comment by antoviaque** [comment]
@bradenmacdonald The idea is to make the progress and discussions fully transparent -- a bit like if we were able to open our Jira board & tickets for this specific project. So it depends on what we put in the tickets on that board. Some questions to figure it out:
- Is there a 1-to-1 match between the tickets on https://github.com/orgs/openedx/projects/4/views/14 and our internal tickets scheduled in our individual sprints? The tickets on the public github board look like individual features/mini-epics, no?
- _If_ there is a 1-to-1 match between our internal tickets and these tickets, are all the discussions we would normally have on our internal Jira tickets held on the corresponding github ticket?

**Comment by bradenmacdonald** [comment]
@antoviaque 

>Is there a 1-to-1 match between the tickets on https://github.com/orgs/openedx/projects/4/views/14 and our internal tickets scheduled in our individual sprints? The tickets on the public github board look like individual features/mini-epics, no?

More or less. We have some minor tasks that I create like 'Touch base with Jenna on Friday.', 'think of some new name ideas for "ALUs" project', 'onboard Jill', etc. but those are more related to internal process than this project.

For example, my only task in the current sprint is "MNG-3388 Contribute technical parts to Approach Spec Memo" which corresponds to my part of https://github.com/openedx/platform-roadmap/issues/209

> If there is a 1-to-1 match between our internal tickets and these tickets, are all the discussions we would normally have on our internal Jira tickets held publicly on the corresponding github ticket?

We aren't having any discussions on the internal Jira tickets, other than about onboarding someone else to the project. All project-related discussions are happening in public on Slack #modular-learning and/or in calls that we have every week or so. For example, the link to the draft approach spec Google Doc was posted on Slack, and that's where all our work is currently happening.

**Comment by antoviaque** [comment]
@bradenmacdonald Thanks for the details! Then it looks like we are already doing everything publicly, which is great :) Kudos for already moving it forward this way. :+1: 

@hurtstotouchfire Does the approach work for you too, and does it matches what you had in mind too?

**Comment by hurtstotouchfire** [comment]
> It would also be useful to be able to point PR authors to public boards that they could check/follow themselves in order to get updates on when to expect a review for their PR. 

@itsjeyd I think it's worth having some synchronous conversations about this. Frankly I would expect that it is not possible to forecast review periods for any repos until they come under Maintainership and within that framework we are only just starting to define the process. We can start with a 1:1 possibly but I think we need to ensure that your efforts are central to the next  maintainership pilot phase.

**Comment by itsjeyd** [comment]
Thanks @hurtstotouchfire, I'd be happy to talk about where OSPR management would fit in with the next maintainership pilot phase.

As I mentioned on Slack, @mphilbrick211 should probably join the synchronous conversation as well.

**Comment by mphilbrick211** [comment]
Thanks @itsjeyd! @hurtstotouchfire let me know how I can help!

**Comment by itsjeyd** [comment]
See [2021-12-09 Office Hours Notes](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3604119553/2021-12-09+Office+Hours+Notes) for the latest status of this conversation.

**Comment by nedbat** [comment]
Changed to blocked, pending phase 2 of maintainership.

**Comment by nedbat** [comment]
Assigned (optimistically) to Q3 2023

**Comment by nedbat** [comment]
@hurtstotouchfire This seems like something that wouldn't happen until Q4?

**Comment by e0d** [comment]
This needs to be groomed and probably broken up into smaller deliverables.  Part of the process should be understanding what's still valuable here and should be our priority.
